src/node/snapshotter.h

```diff
@@ -248,16 +258,34 @@ namespace ccf
       std::lock_guard<std::mutex> guard(lock);
 
       CCF_ASSERT_FMT(
-        idx >= next_snapshot_indices.back(),
+        idx >= next_snapshot_indices.back().idx,
         "Committable seqno {} < next snapshot seqno {}",
         idx,
-        next_snapshot_indices.back());
+        next_snapshot_indices.back().idx);
+
+      bool forced = store->flag_enabled(
+        kv::AbstractStore::Flag::SNAPSHOT_AT_NEXT_SIGNATURE);
+
+      consensus::Index last_unforced_idx = last_snapshot_idx;
+      for (size_t i = next_snapshot_indices.size() - 1;
+           i < next_snapshot_indices.size();
+           i--)
+      {
+        if (!next_snapshot_indices[i].forced)
+        {
+          last_unforced_idx = next_snapshot_indices[i].idx;
+          break;
+        }
+      }
 
-      if ((idx - next_snapshot_indices.back()) >= snapshot_tx_interval)
+      auto due = (idx - last_unforced_idx) >= snapshot_tx_interval;
+      if (due || forced)
       {
-        next_snapshot_indices.push_back(idx);
-        LOG_TRACE_FMT("Recorded {} as snapshot index", idx);
-        return true;
+        next_snapshot_indices.push_back({idx, !due});
```

Comment:
```suggestion
        next_snapshot_indices.push_back({idx, forced});
```

Reply:
No, that's incorrect when `due && forced`. If the two coincide, we can ignore the forced one.